### PR TITLE
Adding Integer translation of metaedges from metapath

### LINKED ISSUE #12: hetio.abbreviation.metaedges_from_metapath breaks for integers
I have been building hetnets for MSigDB at https://github.com/greenelab/interpret-compression and have been using MSigDB collection names for hetio metagraph IDs.

However, I am receiving an error in the function [`metapath_from_abbrev`](https://github.com/dhimmel/hetio/blob/62c4dd1db11733f9e50c980e38a0ccc483aa1546/hetio/hetnet.py#L329). For example, when calling `graph.metagraph.metapath_from_abbrev('GpC1')`

The traceback is:

```python
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-6-2cbc2580bcf8> in <module>()
      1 DWPCs = collections.OrderedDict()
      2 for name, graph in hetnets.items():
----> 3     metapath = graph.metagraph.metapath_from_abbrev('GpC1')
      4     rows, cols, dwpc_matrix, seconds = dwpc(graph, metapath, damping=0.4)
      5     DWPCs[name] = dwpc_matrix

~/anaconda3/envs/interpret-compression/lib/python3.6/site-packages/hetio/hetnet.py in metapath_from_abbrev(self, abbrev)
    333         for metaedge_abbrev in metaedge_abbrevs:
    334             metaedge_id = hetio.abbreviation.metaedge_id_from_abbreviation(
--> 335                 self, metaedge_abbrev)
    336             metaedges.append(self.get_edge(metaedge_id))
    337         return self.get_metapath(tuple(metaedges))

~/anaconda3/envs/interpret-compression/lib/python3.6/site-packages/hetio/abbreviation.py in metaedge_id_from_abbreviation(metagraph, abbreviation)
    128     abbrev_to_kind = {v: k for k, v in metagraph.kind_to_abbrev.items()}
    129     source_kind = abbrev_to_kind[source_abbrev]
--> 130     target_kind = abbrev_to_kind[target_abbrev]
    131     metanode = metagraph.get_node(source_kind)
    132     for edge in metanode.edges:

KeyError: 'C'
```

It appears the root of the error is in in [line 332](https://github.com/dhimmel/hetio/blob/62c4dd1db11733f9e50c980e38a0ccc483aa1546/hetio/hetnet.py#L332) to the method `metaedges_from_metapath`:

```python
hetio.abbreviation.metaedges_from_metapath('GpC1')
>>> ['GpC']
```

Which also appears to break in line 114 in [`hetio/abbreviation.py`](https://github.com/dhimmel/hetio/blob/25628506efab232041740a4a2b1ab7ebcb821dbb/hetio/abbreviation.py#L114).

I have not looked in the specific line in too much detail but I was wondering if there would be a way to accept integers in metapath abbreviations.

```python
pattern = regex.compile('(?<=^|[a-z<>])[A-Z]+[a-z<>]+[A-Z]+')
```

### PR DESCRIPTION
**Pull request by gwaygenomics**
closes #12

### CODE REVIEW

**Comment by gwaygenomics** [comment]
@dhimmel - looks like tests passed. Anything else I should do with this PR? Thanks!

File: hetio/abbreviation.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def metaedges_from_metapath(abbreviation, standardize_by=None):
     if isinstance(standardize_by, hetio.hetnet.MetaGraph):
         metapath = standardize_by.metapath_from_abbrev(abbreviation)
         return [metaedge.get_standard_abbrev() for metaedge in metapath]
-    pattern = regex.compile('(?<=^|[a-z<>])[A-Z]+[a-z<>]+[A-Z]+')
+    pattern = regex.compile('(?<=^|[a-z<>])[A-Z0-9]+[a-z<>]+[A-Z0-9]+')
```
Comment (dhimmel) [inline hetio/abbreviation.py]: Currently this treats all digits as metanode abbreviations... i.e. 0-9 is always uppercase. My idea was require the first character of an abbreviation to be a letter, then allow numbers to follow. The first letter would set the case and thus whether it's a metanode or metaedge.

With the current implementation it is not possible to add numbers to metaedge abbreviations, which could be helpful for some users.

File: test/abbreviation_test.py [hunk 2]
```diff
@@ -53,3 +53,18 @@ def test__find_abbrevs():
...
     for kind_to_abbrev in kind_to_abbrevs:
         assert kind_to_abbrev == find_abbrevs(kind_to_abbrev.keys())
+
+
+def test__metaedges_from_metapath():
+    """
+    Get metaedge subsets from metapath abbreviation
+    """
+    metapath_to_metaedge = {
+            'GpC1': ['GpC1'],
```
Comment (dhimmel) [inline test/abbreviation_test.py]: Let's add a test for `XxXyYyyYzzZZzZZZ`.

File: test/abbreviation_test.py [hunk 3]
```diff
@@ -53,3 +53,18 @@ def test__find_abbrevs():
...
+            'GiGpBP': ['GiG', 'GpBP'],
+            'G1Fp1C2pC1': ['G1Fp1C2', '1C2pC1']
```
Comment (dhimmel) [inline test/abbreviation_test.py]: How I was thinking about this was that G1Fp1C2pC1 would be G1Fp1C2, C2pC1

File: hetio/abbreviation.py (hunk 1 above)
Comment (gwaygenomics) [reply hetio/abbreviation.py]: is it sufficient to check and disallow if first element of metanode is a digit? 07027e2

Comment (dhimmel) [reply hetio/abbreviation.py]: I would prefer for the regex to be aware that metanodes cannot start with a digit. How about changing the node regex to `[A-Z][A-Z0-9]*`? I think the consistency will be better in the longrun.

File: test/abbreviation_test.py [hunk 4]
```diff
@@ -53,3 +53,18 @@ def test__find_abbrevs():
...
+
+
+def test__metaedges_from_metapath():
+    """
+    Get metaedge subsets from metapath abbreviation
+    """
+    metapath_to_metaedge = {
+            'GpC1': ['GpC1'],
+            'GiGpBP': ['GiG', 'GpBP'],
+            'XxXyYyyYzzZZzZZZ': ['XxX', 'XyY', 'YyyY', 'YzzZZ', 'ZZzZZZ']
```
Comment (dhimmel) [inline test/abbreviation_test.py]: Let's add test for `X111yX2zY3Y3Y`

File: hetio/abbreviation.py [hunk 5]
```diff
@@ -37,13 +37,28 @@ def validate_abbreviations(metagraph):
         if not abbrev.isupper():
             print('lowercase metanode abbreviation:', abbrev)
             valid = False
+
+    # Check if metanode abbreviation starts with a digit
```
Comment (dhimmel) [inline hetio/abbreviation.py]: I have a small preference that these checks are integrated into the for loop above. The existing comments like 

```python
     # Check capitalizations
     # metanode abbreviations should be uppercase
```

would be moved to the appropriate inner loop location.

File: hetio/abbreviation.py [hunk 6]
```diff
@@ -111,7 +121,8 @@ def metaedges_from_metapath(abbreviation, standardize_by=None):
     if isinstance(standardize_by, hetio.hetnet.MetaGraph):
         metapath = standardize_by.metapath_from_abbrev(abbreviation)
         return [metaedge.get_standard_abbrev() for metaedge in metapath]
-    pattern = regex.compile('(?<=^|[a-z<>])[A-Z]+[a-z<>]+[A-Z]+')
+    regex_string = '(?<=^|[a-z<>])[A-Z][A-Z0-9]*+[a-z<>]+[A-Z][A-Z0-9]*+'
```
Comment (dhimmel) [inline hetio/abbreviation.py]: I think there are extra `+` after `*`... do you agree?

**Pull request merged**